src/Persistence/Reflection/EnumReflectionProperty.php

```diff
@@ -60,9 +63,40 @@ public function getValue($object = null)
     public function setValue($object, $value = null): void
     {
         if ($value !== null) {
-            $value = $this->enumType::from($value);
+            $value = $this->toEnum($value);
         }
 
         $this->originalReflectionProperty->setValue($object, $value);
     }
+
+    /**
+     * @param BackedEnum|BackedEnum[] $enum
+     *
+     * @return ($enum is BackedEnum ? (string|int) : (string[]|int[]))
+     */
+    private function fromEnum($enum)
+    {
+        if (is_array($enum)) {
+            /** @var string[]|int[] $values */
+            $values = array_map([self::class, 'fromEnum'], $enum);
+
+            return $values;
```

Comment:
Let's avoid recursion in favour of something along

```suggestion
            return array_map(fn(BackedEnum $e) => $e->value, $enum);
```

Reply:
But compatible with PHP 7.2 :(

Reply:
Done